commit 7b302c57ed0db48b688c11156bda78358b083c99
Author: Ed Schouten <ed@nuxi.nl>
Date:   Tue Sep 10 16:39:55 2024 +0200

    Use the go_grpc_v2 compiler for generating gRPC service bindings
    
    By default, Gazelle will use use the ancient go_grpc compiler for
    generating Protobuf and gRPC bindings. What's inconvenient about that is
    that those bindings don't use the grpc.ServiceRegistrar interface that
    was added back in 2020:
    
    https://github.com/grpc/grpc-go/issues/3966
    
    Instead, it uses concrete type *grpc.Server. In Buildbarn I need to make
    use of the interface to support things like xDS, which grpc-go only
    offers via a different concrete server type.
    
    In the past I never had to care about this, because I would regenerate
    these sources myself. Now that I want to remote-apis through bzlmod,
    this is less practical.

diff --git build/bazel/remote/asset/v1/BUILD build/bazel/remote/asset/v1/BUILD
index 5127167..6f050ca 100644
--- build/bazel/remote/asset/v1/BUILD
+++ build/bazel/remote/asset/v1/BUILD
@@ -41,7 +41,10 @@ cc_grpc_library(
 # Go
 go_proto_library(
     name = "remote_asset_go_proto",
-    compilers = ["@io_bazel_rules_go//proto:go_grpc"],
+    compilers = [
+        "@io_bazel_rules_go//proto:go_proto",
+        "@io_bazel_rules_go//proto:go_grpc_v2",
+    ],
     importpath = "github.com/bazelbuild/remote-apis/build/bazel/remote/asset/v1",
     proto = ":remote_asset_proto",
     deps = [
diff --git build/bazel/remote/execution/v2/BUILD build/bazel/remote/execution/v2/BUILD
index 073b792..b48e027 100644
--- build/bazel/remote/execution/v2/BUILD
+++ build/bazel/remote/execution/v2/BUILD
@@ -44,7 +44,10 @@ cc_grpc_library(
 # Go
 go_proto_library(
     name = "remote_execution_go_proto",
-    compilers = ["@io_bazel_rules_go//proto:go_grpc"],
+    compilers = [
+        "@io_bazel_rules_go//proto:go_proto",
+        "@io_bazel_rules_go//proto:go_grpc_v2",
+    ],
     importpath = "github.com/bazelbuild/remote-apis/build/bazel/remote/execution/v2",
     proto = ":remote_execution_proto",
     deps = [
diff --git build/bazel/remote/logstream/v1/BUILD build/bazel/remote/logstream/v1/BUILD
index 70db59d..8dc491c 100644
--- build/bazel/remote/logstream/v1/BUILD
+++ build/bazel/remote/logstream/v1/BUILD
@@ -34,7 +34,10 @@ cc_grpc_library(
 # Go
 go_proto_library(
     name = "remote_logstream_go_proto",
-    compilers = ["@io_bazel_rules_go//proto:go_grpc"],
+    compilers = [
+        "@io_bazel_rules_go//proto:go_proto",
+        "@io_bazel_rules_go//proto:go_grpc_v2",
+    ],
     importpath = "github.com/bazelbuild/remote-apis/build/bazel/remote/logstream/v1",
     proto = ":remote_logstream_proto",
 )
